tests/e2e/collectives/CMakeLists.txt

```diff
@@ -114,5 +117,6 @@ if(IREE_TARGET_BACKEND_ROCM AND IREE_HAL_DRIVER_HIP AND IREE_HIP_TEST_TARGET_CHI
     LABELS
       ${COMMON_LABELS}
       "requires-multiple-devices"
+      "requires-4-devices"
```

Comment:
Did you want these tests running in the `test_amd_mi250` job? I don't see them there yet: https://github.com/iree-org/iree/actions/runs/9140284637/job/25133555370?pr=17433

May need to plumb labels or other features through the workflow and test script: https://github.com/iree-org/iree/blob/a3b74bcc7bd29958c8188e3b79d7000de3427786/build_tools/cmake/ctest_all.sh#L100-L102 https://github.com/iree-org/iree/blob/a3b74bcc7bd29958c8188e3b79d7000de3427786/.github/workflows/ci.yml#L393-L430

Similar story for `test_amd_w7900`, when it is re-enabled (https://github.com/iree-org/iree/issues/17370)